lib/blocs/providers/auth_notifier.dart

```diff
@@ -35,67 +33,58 @@ class LoginNotLogged extends LoginStates {
 
 class AuthNotifier extends StateNotifier<LoginStates> {
   static final provider = StateNotifierProvider<AuthNotifier, LoginStates>(
-      (ref) => AuthNotifier(ref.read(AuthRepository.provider)),);
+    (ref) => AuthNotifier(
+        ref.read(AuthRepository.provider), ref.read(IUserApi.provider),),
+  );
 
-  AuthNotifier(this._authRepository) : super(LoginInitial());
+  AuthNotifier(this._authRepository, this._iuserApi) : super(LoginInitial());
   final AuthRepository _authRepository;
+  final IUserApi _iuserApi;
 
   Future<void> check() async {
-
-    final ShopLoginModel? loginModelCheck = await _authRepository.loginModelCheck() ;
-    if (loginModelCheck != null ) {
+    final ShopLoginModel? loginModelCheck =
+        await _authRepository.loginModelCheck();
+    if (loginModelCheck != null) {
       state = LoginSuccess(loginModelCheck);
     } else {
       state = LoginNotLogged();
     }
   }
 
-   Future<void> login({
-     required String email,
-     required String password,
-   }) async {
-     try {
-       state = LoginLoading();
-       final loginResponse = await DioHelper.postData(url: EndPoint.login, data: {
-         "email": email,
-         "password": password,
-       },);
-       final json = loginResponse.data as Map<String, dynamic>;
-       if (json['status'] == false) throw StateError('failed');
-       final loginModel = ShopLoginModel.fromJson(json);
-       _authRepository.save(loginModel);
-       state = LoginSuccess(loginModel);
-    } on DioError {
-       state = LoginError("Couldn't fetch data. Is the device online?");
+  Future<void> login({
+    required String email,
+    required String password,
+  }) async {
+    state = LoginLoading();
+    try {
+      final dataLogin = await _iuserApi.login(email: email, password: password);
+      _authRepository.save(dataLogin);
+      state = LoginSuccess(dataLogin);
+    } catch (e) {
+      state = LoginError((e as StateError).message);
     }
-   }
+  }
 
   Future<void> register({
     required String name,
     required String email,
     required String password,
     required String phone,
   }) async {
+    state = LoginLoading();
     try {
-      state = LoginLoading();
-      final loginResponse = await DioHelper.postData(url: EndPoint.register, data: {
-        "name": name,
-        "email": email,
-        "password": password,
-        "phone": phone
-      },);
-      final json = loginResponse.data as Map<String, dynamic>;
-      if (json['status'] == false) throw StateError('failed');
-      final loginModel = ShopLoginModel.fromJson(json);
-      _authRepository.save(loginModel);
-      state = LoginSuccess(loginModel);
-    } on DioError {
-      state = LoginError("Couldn't fetch data. Is the device online?");
+      final dataRegister = await _iuserApi.register(
+          name: name, email: email, password: password, phone: phone,);
+      _authRepository.save(dataRegister);
```

Comment:
save is a future void must await for it to ensure its do its logic, and same for the authRepostiroy save function